pkg/log/zap/zap_test.go

```diff
@@ -296,3 +300,128 @@ var _ = Describe("Zap logger setup", func() {
 		})
 	})
 })
+
+var _ = Describe("Zap flag options setup", func() {
+	fromFlags := Options{}
+
+	BeforeEach(func() {
+		flag.CommandLine = flag.NewFlagSet(os.Args[0], flag.ExitOnError) //flags are now reset
+	})
+
+	Context("with -zap-devel options provided", func() {
+		It("Should set dev=true with relative options.", func() {
+			os.Args = []string{"cmd", "-zap-devel=true"}
+			fromFlags.BindFlags(flag.CommandLine)
+			flag.Parse()
+			out := Options{}
+			UseFlagOptions(&fromFlags)(&out)
+			out.addDefaults()
```

Comment:
Non-blocking nit: this unit test is testing both `UseFlagOptions` and `addDefaults()`. Could we skip the `out.addDefaults` call here, and then below, only check that `out.Development` is `true` since the other checks are checking what `addDefaults` does?

And then add a couple of tests for `addDefaults` where we're setting up the `Options` struct manually instead of parsing flags. That way we separate the concerns a bit.

Ditto in other tests that are testing just the flag parsing into options.